### PR TITLE
[BEAM-25] Some more RunnableOnService tests for stateful ParDo

### PR DESCRIPTION
**Pull request by kennknowles**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

Opening a PR to easily launch Jenkins builds. Will likely add more tests before review.

### CODE REVIEW

**Comment by kennknowles** [comment]
https://builds.apache.org/view/Beam/job/beam_PostCommit_Java_RunnableOnService_Dataflow/1985/

**Comment by kennknowles** [comment]
R: @bjchambers 

We'll want more, but thought I would submit early and often. Any particular suggestions based on these?

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Mean.java [hunk 1]
```diff
@@ -101,7 +101,7 @@ private Mean() { } // Namespace only
 
   /////////////////////////////////////////////////////////////////////////////
 
-  private static class MeanFn<NumT extends Number>
+  static class MeanFn<NumT extends Number>
```
Comment (dhalperi) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Mean.java]: necessary? or can you use [`Mean.<Double>of()`](https://github.com/apache/beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Mean.java#L98) in the one place you do `new MeanFn<Double>()`?

Comment (kennknowles) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Mean.java]: I can use the thing you suggest, and certainly will. Thanks!

**Pull request merged**